src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -4409,8 +4409,13 @@ public void setEarlyDialogTimeoutSeconds(int seconds) {
     public void checkRetransmissionForForking(SIPResponse response) {
         final int statusCode = response.getStatusCode();
         final String responseMethod = response.getCSeqHeader().getMethod();
-        final long responseCSeqNumber = response.getCSeq().getSeqNumber();   
-        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod);            
+        final long responseCSeqNumber = response.getCSeq().getSeqNumber();
+        RSeq rseq = (RSeq) response.getHeader(RSeqHeader.NAME);
+        String responseRSeqNumber = "";
```

Comment:
cant we just compose the intial string here, and then add RSeq part just if necessary.
`
 String retransCondition =(statusCode + "/" + responseCSeqNumber + "/" + responseMethod)
 if(rseq != null) {
            retransCondition = retransCondition  + "/" + rseq.getSeqNumber();
        }
boolean isRetransmission = !responsesReceivedInForkingCase.add(retransCondition)
`